File: backend/app/schemas/match_result.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class MatchResultCreate(BaseModel):
    resume_id: int
    job_posting_id: int
    overall_score: float
    strengths: list[Any] | None = None
    weaknesses: list[Any] | None = None
    missing_keywords: list[Any] | None = None
    suggestions: list[Any] | None = None


class MatchResultUpdate(BaseModel):
    overall_score: float | None = None
    strengths: list[Any] | None = None
    weaknesses: list[Any] | None = None
    missing_keywords: list[Any] | None = None
    suggestions: list[Any] | None = None


class MatchResultListItem(BaseModel):
    id: int
    resume_id: int
    job_posting_id: int
    overall_score: float
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class MatchResultRead(MatchResultListItem):
    strengths: list[Any] | None
    weaknesses: list[Any] | None
    missing_keywords: list[Any] | None
    suggestions: list[Any] | None

File: frontend/src/types/match_result.ts
import type { ISODateString, JsonValue } from "./common";

export interface MatchResultListItem {
  id: number;
  resume_id: number;
  job_posting_id: number;
  overall_score: number;
  created_at: ISODateString;
}

export interface MatchResult extends MatchResultListItem {
  strengths: JsonValue[] | null;
  weaknesses: JsonValue[] | null;
  missing_keywords: JsonValue[] | null;
  suggestions: JsonValue[] | null;
}

export interface MatchResultCreate {
  resume_id: number;
  job_posting_id: number;
  overall_score: number;
  strengths?: JsonValue[] | null;
  weaknesses?: JsonValue[] | null;
  missing_keywords?: JsonValue[] | null;
  suggestions?: JsonValue[] | null;
}

export type MatchResultUpdate = Partial<
  Pick<
    MatchResultCreate,
    "overall_score" | "strengths" | "weaknesses" | "missing_keywords" | "suggestions"
  >
>;

export interface MatchAnalysisRequest {
  resume_id: number;
  job_posting_id: number;
}
